src/controllers/helpers/pagination.rs

```diff
@@ -14,19 +14,13 @@ pub(crate) enum Page {
 }
 
 impl Page {
-    fn new(params: &IndexMap<String, String>) -> AppResult<Self> {
-        if let Some(s) = params.get("page") {
-            let numeric_page = s.parse().map_err(|e| bad_request(&e))?;
-            if numeric_page < 1 {
-                return Err(bad_request(&format_args!(
-                    "page indexing starts from 1, page {} is invalid",
-                    numeric_page,
-                )));
+    fn new(page: Option<u32>) -> Page {
+        match page {
+            Some(page) => {
+                assert_ne!(page, 0);
```

Comment:
I haven't reviewed closely, but this caught my eye. We try to avoid potential panics in the request/response cycle. This refactor looks fine, but I suggest you add a doc comment with a `# Panics` section stating that the caller is responsible for ensuring this.